src/kernel-install/90-loaderentry.install.in

```diff
@@ -153,7 +153,7 @@ mkdir -p "${LOADER_ENTRY%/*}" || {
     echo "linux      $KERNEL_ENTRY"
 
     have_initrd=
-    for initrd in "${@}" "${KERNEL_INSTALL_STAGING_AREA}"/initrd*; do
+    for initrd in $(printf '%s\n' "${@}" "${KERNEL_INSTALL_STAGING_AREA}"/initrd* | sort -Vu); do
```

Comment:
I'm not entirely sure about this change, since it's a bit of a backward incompatible change. The things passed on the commandline would have always come first before and would not get sorted in. Also, at least for the simple cases like `initrd-00-foo` and `initrd-10-bar` mentioned, they would already come out already sorted form the glob because of their lexicographical order, no?

I don think that there's room here for a change, though, to make it easier to ensure that microcode comes first. Maybe something like 
```suggestion
    for initrd in "${KERNEL_INSTALL_STAGING_AREA}"/early-initrd* "${@}" "${KERNEL_INSTALL_STAGING_AREA}"/initrd*; do
```
so that plugins could hook into this by staging all the microcode they want to load. I'm neither sure about the order with respect to `$@` nor the name, though.